rules/template-indent.js

```diff
@@ -0,0 +1,135 @@
+'use strict';
+const stripIndent = require('strip-indent');
+
+const MESSAGE_ID_IMPROPERLY_INDENTED_TEMPLATE = 'template-indent';
+const MESSAGE_ID_INVALID_NODE_TYPE = 'invalid-node-type';
+const messages = {
+	[MESSAGE_ID_IMPROPERLY_INDENTED_TEMPLATE]: 'Templates should be properly indented. Selector: {{ selector }}',
+	[MESSAGE_ID_INVALID_NODE_TYPE]: 'Invalid node type matched by selector {{ selector }}. Expected TemplateLiteral, got {{ type }}',
+};
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => {
+	const options = {
+		tags: ['outdent', 'dedent', 'gql', 'sql', 'html', 'styled'],
+		functions: ['dedent', 'stripIndent'],
+		selectors: [],
+		...context.options[0],
+	};
+
+	const selectors = [
+		...options.tags.map(tag => `TaggedTemplateExpression[tag.name="${tag}"] > .quasi`),
+		...options.functions.map(fn => `CallExpression[callee.name="${fn}"] > .arguments:first-child`),
+		...options.selectors,
+	];
+
+	/** @param {string} selector */
+	const getTemplateLiteralHandler = selector =>
+		/** @param {import('@babel/core').types.TemplateLiteral} node */
+		node => {
+			if (node.type !== 'TemplateLiteral') {
+				context.report({
+					node,
+					messageId: MESSAGE_ID_INVALID_NODE_TYPE,
+					data: {
+						selector,
+						type: node.type,
+					},
+				});
+				return;
+			}
```

Comment:
Let's just ignore it.

User can use `CallExpression[callee.name="foo"] > .arguments`, but `foo` may also accept other types, so we only check

```js
foo(`
	// A template string
`)
```

and ignore

```js
foo('A string not a template')
```

User don't have to use `CallExpression[callee.name="foo"] > TemplateLiteral.arguments` selector.